src/main/scala/co/teapot/tempest/algorithm/MonteCarloPPR.scala

```diff
@@ -23,26 +23,41 @@ import scala.concurrent.{Await, Future}
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.util.Random
 
+
 /**
   * Performing Monte Carlo walks from a list of seed nodes (specified by their ids)
   */
 object MonteCarloPPR {
-  def estimatePPR(graph: DirectedGraph, startIds: IndexedSeq[Int], params: MonteCarloPageRankParams,
+  def estimatePPR(graph: DirectedGraph,
+                  startIds: IndexedSeq[Int],
+                  firstStepDirection: EdgeDir,
```

Comment:
Good call. New comment:
```
/**
  * Performing Monte Carlo walks from a list of seed nodes (specified by their ids).
  * This method is generalized in a few different ways:
  *   - LengthConstraint specifies that the walk have even, odd, or any length.
  *   - firstStepDirection can specify an out-neighbor or in-neighbor is followed first.
  *   - params.alternating specifies whether the walk keeps following out-neighbors (PPR) or alternates between out
  *     and in-neighbors (SALSA)
  *  As examples, on Twitter if you have a tweet producer like Obama and want to find similar tweet producers,
  *     you likely want to use an even-length alternating walk, starting with EdgeDir.In (to first find followers of Obama).
  *  On Twitter if you have a tweet consumer, say Alice, and want to find similar tweet consumers,
  *     you likely want to use an even-length alternating walk, starting with EdgeDir.Out (to first find who Alice follows).
  *  On Twitter if you have a tweet consumer, say Alice, and want to recommend who-to-follow,
  *     you likely want to use an odd-length alternating walk, starting with EdgeDir.Out (to first find who Alice follows).
  *  On Twitter, if we want to find a mega-celbrity, who other celebrities follow, you likely want to use a non-alternating
  *     walk of unconstrained length
  *  On a user-item bipartite graphs, if you want to find similar items to a given item, see the tweet producer case above.
  *  On a user-item bipartite graphs, if you want to find similar users to a given user, see the tweet consumer case above.
  *  On a user-item bipartite graphs, if you want to recommend items to a given user, see the who-to-follow case above.
  *
  *  Background reading: "WTF: the who to follow service at Twitter" by Gupta, Goel, Lin, Sharma, Wnag, Zadeh
  */
```
I'm now thinking we should have individual methods like `similarProducersMonteCarlo`, `similarConsumersMonteCarlo`, and `recommendProducerMonteCarlo`, probably both in the client and the server. I'll need to rethink the PPR call anyway once we have typed nodes, so I'm going to punt on those methods for now.